selectel/resource_selectel_mks_cluster_v1_test.go

```diff
@@ -271,3 +282,167 @@ func testAccMKSClusterV1Zonal(projectName, clusterName, kubeVersion, maintenance
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVPCV2ProjectExists("selectel_vpc_project_v2.project_tf_acc_test_1", &project),
+					testAccCheckMKSClusterV1DefaultKubeVersionFeatureGates("selectel_vpc_project_v2.project_tf_acc_test_1", &featureGates),
+				),
+			},
+		},
+	})
+
+	return featureGates
+}
+
+func testAccCheckMKSClusterV1DefaultKubeVersionFeatureGates(n string, featureGates *[]string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		mksClient, err := newTestMKSClient(n, s)
+		if err != nil {
+			return err
+		}
+		ctx := context.Background()
+
+		defaultVersion, err := getDefaultKubeVersion(ctx, mksClient)
+		if err != nil {
+			return err
+		}
+
+		allFeatureGates, _, err := kubeoptions.ListFeatureGates(ctx, mksClient)
+		if err != nil {
+			return err
+		}
+		kubeVersionMinor, err := kubeVersionTrimToMinor(defaultVersion)
+		if err != nil {
+			return err
+		}
+
+		for _, item := range allFeatureGates {
+			if kubeVersionMinor == item.KubeVersion {
+				*featureGates = item.Names
+			}
+		}
+
+		return nil
+	}
+}
+
+func testDefaultAdmissionControllers(t *testing.T) []string {
+	var project projects.Project
+	admissionContollers := make([]string, 0)
+	projectName := acctest.RandomWithPrefix("tf-acc")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccSelectelPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckVPCV2ProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVPCV2ProjectBasic(projectName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVPCV2ProjectExists("selectel_vpc_project_v2.project_tf_acc_test_1", &project),
+					testAccCheckMKSClusterV1DefaultKubeVersionAdmissionControllers("selectel_vpc_project_v2.project_tf_acc_test_1", &admissionContollers),
+				),
+			},
+		},
+	})
+
+	return admissionContollers
+}
+
+func testAccCheckMKSClusterV1DefaultKubeVersionAdmissionControllers(n string, admissionControllers *[]string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		mksClient, err := newTestMKSClient(n, s)
+		if err != nil {
+			return err
+		}
+		ctx := context.Background()
+
+		defaultVersion, err := getDefaultKubeVersion(ctx, mksClient)
+		if err != nil {
+			return err
+		}
+
+		allAdmissionControllers, _, err := kubeoptions.ListAdmissionControllers(ctx, mksClient)
+		if err != nil {
+			return err
+		}
+		kubeVersionMinor, err := kubeVersionTrimToMinor(defaultVersion)
+		if err != nil {
+			return err
+		}
+
+		for _, item := range allAdmissionControllers {
+			if kubeVersionMinor == item.KubeVersion {
+				*admissionControllers = item.Names
+			}
+		}
+
+		return nil
+	}
+}
+
+func newTestMKSClient(n string, s *terraform.State) (*v1.ServiceClient, error) {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return nil, fmt.Errorf("not found: %s", n)
+	}
+
+	if rs.Primary.ID == "" {
+		return nil, errors.New("no ID is set")
+	}
```

Comment:
These checks logically are not related to the creation of the MKS client, it's a little confusing. You should not include them in this function.